src/app/task/[id]/page.tsx

```diff
@@ -1,3 +1,173 @@
+"use client";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useSelector } from "react-redux";
+import { useDispatch } from "react-redux";
+import {
+  selectTaskById,
+  selectCompletedSubtasksPercentage,
+} from "../../../redux/features/taskSlice";
+import {
+  handleCompleteSubtask,
+  handleDeleteTask,
+  repeatSubtasks,
+} from "../../../redux/features/taskSlice";
+import { RootState } from "../../../redux/store";
+import { BackIcon } from "../../icons/BackIcon";
+import { CalendarIcon } from "../../icons/CalendarIcon";
+import { PriorityIcon } from "../../icons/PriorityIcon";
+import { levelsCount } from "../../components/LevelSelector";
+import { ComplexityIcon } from "../../icons/ComplexityIcon";
+import { Progress } from "../../components/ui/progress";
+import { CompleteSubtaskIcon } from "../../icons/CompleteSubtaskIcon";
+import { RepeatIcon } from "../../icons/RepeatIcon";
+import Title from "@/app/components/Title";
+
+function determineLevel(value: number): string {
+  if (value < 4) {
+    return "Low";
+  } else if (value < 7) {
+    return "Medium";
+  } else {
+    return "High";
+  }
+}
```

Comment:
You could move this to a utils folder.

Reply:
Ok. It makes sense. I'll move it to utils folder and merge into main.